drivers/coreaudio/audio_driver_coreaudio.cpp

```diff
@@ -142,7 +142,7 @@ Error AudioDriverCoreAudio::init() {
 
 	unsigned int buffer_size = buffer_frames * channels;
 	samples_in.resize(buffer_size);
-	input_buf.resize(buffer_size);
+	input_buf.resize(buffer_size * 2);
```

Comment:
I assume it's to fix:

> a "-50" error from AudioUnitRender in audio_driver_coreaudio.cpp whenever microphone input is enabled. This is because bufferList.mBuffers[0].mDataByteSize is too small, since there is a mismatch between the capture buffer format and the audio device.

Maybe buffer size should be calculated separately form output buffer size, without using `output_latency` and `mix_rate`, these properties are for output. Set to the fixed size or add another property?

Reply:
I agree with your point. " * 2" looks definitely a magic number. I hope someone who knows well this audio part can help to fix it in the near future because I'm not an expert in regards to this. On the other hand, fortunately, this file only has to do with the iPhone devices and this change works well on them, so making changes little by little would be also great.

Reply:
> On the other hand, fortunately, this file only has to do with the iPhone devices and this change works well on them, so making changes little by little would be also great.

This file is also used for macOS, so I guess it should be tested there too just in case.


I also think it would be better to calculate `input_buffer_size` separately or make it into constant/define.

Reply:
Let's revert this and fix it later then.